### PR TITLE
Refactor load generation framework to generate gas with only one transaction

### PR DESCRIPTION
**Pull request by sadhansood**
Tested on the private testnet, the setup for 100 qps now takes less then a minute

### CODE REVIEW

File: crates/sui-benchmark/src/util.rs [hunk 1]
```diff
@@ -20,3 +40,184 @@ pub fn get_ed25519_keypair_from_keystore(
...
+    assert_eq!(created_coins.len(), split_amounts.len());
+    let created_coin_ids: Vec<ObjectID> = created_coins.iter().map(|c| c.0).collect();
+    let recipient_addresses: Vec<SuiAddress> = coin_configs.iter().map(|x| x.address).collect();
+    let verified_tx = make_pay_tx(
+        created_coins,
+        sender,
+        recipient_addresses,
+        split_amounts,
+        updated_gas.0,
+        &gas.2,
```
Comment (lxfind) [inline crates/sui-benchmark/src/util.rs:157]: I think you should just use pay_sui.

Comment (sadhansood) [reply crates/sui-benchmark/src/util.rs]: It seems like the only difference b/w `pay` (which is being used here) and `pay_sui` is that you don't need an extra gas coin to pay for the transaction itself. You still need to split the coin into multiple coins which can be transferred to the recipients - so in the end you still need two transactions? This is my understanding but I could be wrong.

Comment (lxfind) [reply crates/sui-benchmark/src/util.rs]: No you don't need to split them. pay_sui splits the coins out for you.

Comment (sadhansood) [reply crates/sui-benchmark/src/util.rs]: Adding TODO for it, will take it up as a follow up. Thanks

File: crates/sui-benchmark/src/bin/stress.rs [hunk 2]
```diff
@@ -374,23 +390,72 @@ async fn main() -> Result<()> {
...
+                            ))
+                        };
+                        let (
+                            shared_counter_workload_init_gas_config,
+                            shared_counter_workload_payload_gas_config,
+                        ) = all_shared_counter_coin_configs.unwrap_or((vec![], vec![]));
+                        let (
+                            transfer_object_workload_tokens,
+                            transfer_object_workload_payload_gas_config,
+                        ) = all_transfer_object_coin_configs.unwrap_or((vec![], vec![]));
```
Comment (lxfind) [inline crates/sui-benchmark/src/bin/stress.rs:433]: Instead of unwrap_or here, why not just return empty vectors above when in the if branches?

**Review by lxfind — COMMENTED** [review]
Could you break down the main function to sub-functions? It's now 400 lines..

File: crates/sui-benchmark/src/bin/stress.rs (hunk 2 above)
Comment (sadhansood) [reply crates/sui-benchmark/src/bin/stress.rs]: fixed

File: crates/sui-benchmark/src/bin/stress.rs [hunk 3]
```diff
@@ -230,11 +233,12 @@ async fn main() -> Result<()> {
...
         let (owner, keypair): (SuiAddress, AccountKeyPair) = deterministic_random_account_key();
-        let primary_gas = generate_test_gas_objects_with_owner(1, owner);
-        let primary_gas_id = primary_gas.get(0).unwrap().id();
+        let generated_gas = generate_test_gas_objects_with_owner(2, owner);
+        let primary_gas = generated_gas.get(0).unwrap().clone();
+        let pay_coin = generated_gas.get(1).unwrap().clone();
         // Make the client runtime wait until we are done creating genesis objects
         let cloned_config = configs.clone();
-        let cloned_gas = primary_gas;
+        let cloned_gas = generated_gas;
```
[the remark this answers is not in the document]
Comment (sadhansood) [reply crates/sui-benchmark/src/bin/stress.rs]: fixed

**Comment by sadhansood** [comment]
Refactored and split a bunch of code out into smaller modules

**Review by lxfind — APPROVED** [review]
Thank you!

**Pull request merged**